src/Roslyn.Diagnostics.Analyzers/Core/SpecifyDiagnosticDescriptorCustomTags.cs

```diff
@@ -0,0 +1,58 @@
+namespace Roslyn.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class SpecifyDiagnosticDescriptorCustomTags : DiagnosticAnalyzer
+    {
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(RoslynDiagnosticsAnalyzersResources.SpecifyDiagnosticDescriptorCustomTagsTitle), RoslynDiagnosticsAnalyzersResources.ResourceManager, typeof(RoslynDiagnosticsAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(RoslynDiagnosticsAnalyzersResources.SpecifyDiagnosticDescriptorCustomTagsMessage), RoslynDiagnosticsAnalyzersResources.ResourceManager, typeof(RoslynDiagnosticsAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RoslynDiagnosticIds.SpecifyDiagnosticDescriptorCustomTagsRuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.RoslynDiagnosticsUsage,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: false,
+                                                                             helpLinkUri: null,
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                if (compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(typeof(DiagnosticDescriptor).FullName, out var diagnosticDescriptorType))
+                {
+                    compilationContext.RegisterOperationAction(opContext =>
+                    {
+                        var objectCreation = (IObjectCreationOperation)opContext.Operation;
+                        if (objectCreation.Type.Equals(diagnosticDescriptorType) && IsMissingCustomTagsArgument(objectCreation.Arguments))
+                        {
+                            opContext.ReportDiagnostic(objectCreation.CreateDiagnostic(Rule));
+                        }
+                    }, OperationKind.ObjectCreation);
+                }
+            });
+        }
+
+        private static bool IsMissingCustomTagsArgument(ImmutableArray<IArgumentOperation> arguments) =>
```

Comment:
I am not sure whether this is the best way to handle the value passed as argument. Please advise for a better option if any.